util/export_xds_ascii.py

```diff
@@ -13,21 +21,43 @@ def export_xds_ascii(integrated_data, experiment_list, params, var_model=(1, 0))
     """Export data from integrated_data corresponding to experiment_list to
     an XDS_ASCII.HKL formatted text file."""
 
-    from dials.array_family import flex
-
-    # for the moment assume (and assert) that we will convert data from exactly
-    # one lattice...
-
-    assert len(experiment_list) == 1
-    # select reflections that are assigned to an experiment (i.e. non-negative id)
-
-    integrated_data = integrated_data.select(integrated_data["id"] >= 0)
-    assert max(integrated_data["id"]) == 0
-
+    if len(experiment_list) == 1:
+        experiment_data = integrated_data.select(integrated_data["id"] >= 0)
+        _export_experiment(
+            params.xds_ascii.hklout,
+            experiment_data,
+            experiment_list[0],
+            params,
+            var_model,
+        )
+    else:
+        for i, experiment in enumerate(experiment_list):
+            experiment_data = integrated_data.select(integrated_data["id"] == i)
+            name, ext = os.path.splitext(params.xds_ascii.hklout)
+            filename = name + "_{}".format(i) + ext
+            _export_experiment(filename, experiment_data, experiment, params, var_model)
+
+
+def _export_experiment(filename, integrated_data, experiment, params, var_model=(1, 0)):
+    # type: (str, flex.reflection_table, dxtbx.model.Experiment, phil_scope, Tuple)
+    """Export a single experiment to an XDS_ASCII.HKL format file.
+
+    Args:
+        filename: The file to write to
+        integrated_data: The reflection table, pre-selected to one experiment
+        experiment: The experiment list entry to export
+        params: The PHIL configuration object
+        var_model:
+    """
     # export for xds_ascii should only be for non-scaled reflections
     assert any(
         [i in integrated_data for i in ["intensity.sum.value", "intensity.prf.value"]]
     )
+    # Handle requesting profile intensities (default via auto) but no column
+    if "profile" in params.intensity and not "intensity.prf.value" in integrated_data:
```

Comment:
```suggestion
    if "profile" in params.intensity and "intensity.prf.value" not in integrated_data:
```

Reply:
note that I've changed all those on master today.

Reply:
Right - this was new code on the branch anyway, but I forgot that pre-commit flake8 wasn't the complete list.